kernel/poll.c

```diff
@@ -675,7 +675,7 @@ int k_work_poll_submit_to_queue(struct k_work_q *work_q,
 	__ASSERT(work_q != NULL, "NULL work_q\n");
 	__ASSERT(work != NULL, "NULL work\n");
 	__ASSERT(events != NULL, "NULL events\n");
-	__ASSERT(num_events > 0, "zero events\n");
+	__ASSERT(num_events >= 0, "zero events\n");
```

Comment:
Isn't the message now a bit misleading as we allow zero events, should this be something like `invalid event count` or similar?